spec/data-types-and-values.html

```diff
@@ -7,12 +7,63 @@ <h1>ECMAScript Language Types</h1>
 
     <emu-clause id="sec-ecmascript-language-types-record-type">
       <h1>The Record Type</h1>
-      <p>TBD</p>
+      <p>
+        The Record type is the set of all finite mappings from Strings to ECMAScript primitive values including Record and Tuple. Each record value holds an associated [[Fields]] List value which is a list of pairs of the form { [[Key]], [[Value]] } where the [[Key]] is a String and [[Value]] is any primitive value. Entries of [[Fields]] are sorted by [[Key]] in code unit order. The [[Fields]] List and its entries are never modified.
+     </p>
+
+      <emu-clause id="sec-makerecordvalue" aoid="MakeRecordValue">
+        <h1>MakeRecordValue ( _entries_ )</h1>
+        <p>This abstract operation MakeRecordValue takes a _entries_ argument. It is used to specify the creation of new Record value types. It performs the following steps when called:</p>
+        <emu-alg>
+            1. Assert: _entries_ is a List.
+            1. Let _uniqueEntries_ be an empty Set.
+            1. For each _kv_ in _entries_:
+              1. Assert: _kv_.[[Key]] is a String.
+              1. Assert: _kv_.[[Value]] is a primitive type.
+              1. For each _existing_ in _uniqueEntries_:
+                1. If SameValue(_existing_.[[Key]], _kv_.[[Key]]
+                  1. Remove _existing_ from _uniqueEntries_
+                  1. Break

```

Comment:
So far, we've been avoiding using "Break" in spec language, sticking to just early returns. How about instead, you iterate over _entries_ in reverse order, and do "If there is no entry _existing_ in _uniqueEntries_ such that _existing_.[[Key]] is _kv_.[[Key]], append _kv_ to _uniqueEntries_."

Reply:
We don't necessarily need to break, but I will try your approach